### ISSUE TITLE
Suggestion for wordwrap in devices on dashboard

### ISSUE THREAD

**Issue by janpep** [domoticz/domoticz#6085]
I had a problem with the readability of longer lines in text and alert devices.
Under Utility tab there is a wordwrap and text goes to the next line.
Under Dashboard tab this is not the case.
I found a workaround by changing /pathtodomoticz/www/css/styles.css.
See topic: [Incomplete text shown on dashboard](https://www.domoticz.com/forum/viewtopic.php?t=42358)

I guess this shall be overwritten by next update.
Therefore my suggestion to implement this.

```
.status {
        /* 22-05-2024 Jan peppink, Modified to get wordwrap on dashboard.*/
        /*white-space: nowrap;*/
        white-space: normal;
}
```
I can not think of a good reason to have the 'nowrap' that makes the rest of a longer sentence invisible.
Until now I do not see unexpected or unwanted side effects.

**Comment by szczukot** [comment]
I also see this problem
https://www.domoticz.com/forum/viewtopic.php?p=316892

**Comment by gizmocuz** [comment]
Well, it is not meant to display long texts at all.
But the one generating this text could do the word-wrapping?

**Comment by janpep** [comment]
The text comes from RSS feed,
It is called a TEXT device, but should only contain very short text? How short? I could not find the max length. of the text, nor the max lengt of a line. Even one sentence might not fit in current situation, when displayed under the dashboard.
I mention it, because there is a difference in behaviour for the same device (!) when it is shown under Utility and as favorite under Dashboard. That makes me wonder: "Why this difference?" In Uiltity it has the wordwrap in dashboard it has 'nowrap'. (This finding is actually independent of the exact length.)
In my opinion it makes sense that the device should be capable to display all the text that fits in the text field of that same device? I do not know the max. length of the field, but if it can hold xxx characters, I should be able to see all xxx characters in the device.
It does not sound logic that it needs to be a multiline text with lines that is already formatted in a way that do not exceed the width. Also because the widt can vary when playing with the width of the browser.

**Comment by gizmocuz** [comment]
Can't you store it in a uservariable instead? Does a RSS feed need to be displayed?

**Comment by janpep** [comment]
It is only a field that is taken out of the given response by the feed. Not the enrire feed itself.  So, in this case I was asked to put the weatherforcast text in a text device. That is not the problem.
In fact it does not matter where you store it, this is about where and how it can be seen, Fact that it is alreay possible under the Utility tab. So why not under dashboard.

**Comment by janpep** [comment]
In addition: I do not see any downside to removing the 'nowrap'.

**Comment by gizmocuz** [comment]
Are you sure the 'nowrap' is added in the normal dashboard and not for mobile only?
I inspected the styles applied to a text widget in the dashboard and there is no 'nowrap' class added here
Are you using dzVents to download the RSS field? You could wrap here?

**Comment by janpep** [comment]
I found the place to set it in: /pathtodomoticz/www/css/styles.css under '.status.' line 686
On the forum I already showed this part and the line that whas there commented out and replaced,
```
.status {
        /* commented out for text wrap on dashboard.*/
        /*white-space: nowrap;*/
        white-space: normal;
}
```
I use dzVents to download the RSS field. Do not know if I can add an inline field to do the wrapping, but it seems more logic to have that in the CSS then to put it in every text that is inserted. Agiain. What it could be a good reason not to have the NOwrap? As I see it: Only when you prefer to make a part of a sentence unreadable. :-)

**Comment by janpep** [comment]
So when you asked.... "Are you sure the 'nowrap' is added in the normal dashboard and not for mobile only?",  I can say yes. I showed the code that I corrected. Also I mentioned the file and the place. I do not use the mobile setting.

Now going to 2025-2 I still had to replace ....
white-space: nowrap;
... with ...
white-space: normal;
... to get rid of the NOwrap.
I even put it in a script "afterupdate.sh" to restore my change after every update.

Again: The NOwrap serves no purpose other than to ensure that slightly longer lines fall outside the device. May be this can be changed in CSS? Or can the reason that the nowrap is there be explained?

**Comment by gizmocuz** [comment]
@janpep , Are we still talking about text widgets?

Without the above patch (in the default theme) this is my text device

<img width="495" height="193" alt="Image" src="https://github.com/user-attachments/assets/1c6f755a-c85a-4917-80a5-b525e5f20654" />

Seems to wrap just fine

**Comment by janpep** [comment]
And THIS is what I see. And I do not like it this way.
<img width="522" height="154" alt="Image" src="https://github.com/user-attachments/assets/377553d2-dd69-4e62-befb-d83a13949201" />

Unless I make the change in the CSS file.
Where I also remove the icon, but that is not the point here. It is about the text outside over the gray backgrouind.

<img width="411" height="171" alt="Image" src="https://github.com/user-attachments/assets/78f60118-1316-48b6-9472-7617523e64f7" />

**Comment by gizmocuz** [comment]
Could you post the text lines here? Press Edit on the text sensor and copy/paste this here

As you can see in my image above I don't have that text going over the gray, it wraps nicely.

**Comment by janpep** [comment]
Included text is like this. Plain text with `<br>` for the new line like:
`<br>19:00 18 Kerkzaal TEST This is short.<br>20:00 21 Kerkzaal TEST This is a longer text.<br>21:00 12 Kerkzaal TEST And this is even more text that should really wrap the line.<br><br>`

For this theme another path to css file is used than I mentioned before.

<img width="407" height="215" alt="Image" src="https://github.com/user-attachments/assets/5837cfcb-9212-4a72-8b3d-110d64232995" />

**Comment by gizmocuz** [comment]
Default theme:

<img width="484" height="220" alt="Image" src="https://github.com/user-attachments/assets/21ac3fb7-b74d-4a10-8125-eedcdca402be" />

Simple gray:

<img width="499" height="224" alt="Image" src="https://github.com/user-attachments/assets/2eedc059-506c-4593-aca3-88928d88ca1f" />

I just pressed 'Edit' on the Text Sensor, and pasted your above text

Edit: Watching on my desktop with the 'Edge' browser
Edit2: Tested with all included themes, all display correctly
Edit3: Watching in the Utility tab
Edit4: Ahhh my bad! It is going wrong on the dashboard!

**Comment by gizmocuz** [comment]
I have made the patch in beta 16845.
Now there was obvious a reason why I added the no-wrap here, but I can't seem any device in my system that displays wrong until now.

Sorry about the mix up... I tested against the 'Utility' tab and not on the home-screen.
The utility tab also uses the class 'status', but somehow it keeps wrapping here.
The difference is that on the homescreen the text is inside a 'span' object, and in the utility tab it is directly inside the 'td' object.

**Comment by janpep** [comment]
Please refer again to the opening message where I tried to clarify the difference between the utility and dashboard as clearly as possible.
It is now clear that that's where the confusion lay. I briefly thought: "He doesn't WANT to understand." :-)
I am glad that's cleared up now. **Thank you very much!**

**Comment by gizmocuz** [comment]
Yeah maybe next time write it like >>>>>>>>>>> DASHBOARD <<<<<<<<< hehe
